main.py

```
@@ -114,6 +124,58 @@ def perform_addition(self) -> None:
+        image_label.setAlignment(Qt.AlignCenter) # type: ignore
+        layout.addWidget(image_label)
+
+        # --- Logic for loading image ---
+        def load_image_from_url():
+            url = url_input.text()
+            if not validators.url(url):
+                image_label.setText("Invalid URL format.")
+                return
+
+            try:
+                response = requests.get(url, stream=True, timeout=5) # Added timeout
```

Flake8 (7.2.0)

[error] at least two spaces before inline comment

(E261)